File: sand/labs/lab6/lab6.R

library(R.matlab)
data <- readMat('data_v1_binned_static.mat')


dataset_number = 8
stimulus_conditions = data$data[[dataset_number]][[1]][[3]]

# conditions to compare
condition_1 = 2
condition_2 = 10

# neuron x conditions x time x trials
spikes = data$data[[dataset_number]][[1]][[6]][,,,]

# responses to conditions summed over time
spikes_1 = spikes[,condition_1,,]
spikes_2 = spikes[,condition_2,,]

X_1 = t(apply(spikes_1, MARGIN=c(1,3), sum))
X_2 = t(apply(spikes_2, MARGIN=c(1,3), sum))
X = rbind(X_1,X_2)
y = c(rep(0,each=dim(X_1)[1]),rep(1,each=dim(X_2)[1]))
image(X)


# logistic regression
model <- glm(y~X, family = binomial(link = "logit"))
fitted.results <- predict(model,type='response')
fitted.results <- ifelse(fitted.results > 0.5,1,0)
accuracy <- mean(fitted.results == y)
print(accuracy)

File: sand/labs/lab6/lab6.m

% data{dataset_number}.spikes is [neurons x stimulus_conditions x time x repeated_trials]
% data{dataset_number}.conditions contains information about the trial orientation/contrast

dataset_number = 8;

cond1 = 2;  % orientation=0  contrast=100
cond2 = 10; % orientation=90 contrast=100
X1 = squeeze(data{dataset_number}.spikes(:,cond1,:,:));
X2 = squeeze(data{dataset_number}.spikes(:,cond2,:,:));

% total numbers of spikes during the trial
total_X1 = squeeze(sum(X1,2));
total_X2 = squeeze(sum(X2,2));


%% Lab 6.1 - Visualize predictors
% Our goal is to take the activity of all neurons (summed over some time
% range) and discriminate the trials for two specific conditions. We will
% need a [trials x neurons] matrix X that contains the neural activity as
% well as a [trials] vector y that contains the condition identity (0 or 1).

X = ???
y = ???
imagesc(???)

%% Lab 6.2 - Logistic regression

b = glmfit(???); % Fit

%% Lab 6.3 -- Calculate accuracy

yhat = glmval(???); % Predict
acc = ??? % Accuracy

%% 